test/e2e/konnect_test.go

```diff
@@ -319,3 +325,66 @@ func requireKonnectNodesConsistentWithK8s(ctx context.Context, t *testing.T, env
 		return true
 	}, konnectNodeRegistrationTimeout, konnectNodeRegistrationCheck)
 }
+
+// requireAllProxyReplicasIDsConsistentWithKonnect ensures that all proxy replicas are registered in Konnect's Node API with
+// their respective IDs.
```

Comment:
This might be enough but I feel we might add a bit more details here to the "definition" of consistent. Would you mind extending this comment?